src/platform_linux/platform.c

```diff
@@ -56,38 +56,46 @@ void
 platform_heap_destroy(platform_heap_handle UNUSED_PARAM(*heap_handle))
 {}
 
+/*
+ * platform_buffer_create() - Create large buffers using mmap()
+ *
+ * Certain modules, e.g. the buffer cache, need a very large buffer which
+ * may not be serviceable by the heap. Create the requested buffer using
+ * mmap() and return a handle to it.
+ */
 buffer_handle *
-platform_buffer_create(size_t               length,
-                       platform_heap_handle UNUSED_PARAM(heap_handle),
-                       platform_module_id   UNUSED_PARAM(module_id))
+platform_buffer_create(size_t             length,
+                       platform_heap_id   heap_id,
+                       platform_module_id UNUSED_PARAM(module_id))
 {
-   buffer_handle *bh = TYPED_MALLOC(platform_get_heap_id(), bh);
+   buffer_handle *bh = TYPED_MALLOC(heap_id, bh);
+   if (bh == NULL) {
+      return bh;
+   }
 
-   if (bh != NULL) {
-      int prot  = PROT_READ | PROT_WRITE;
-      int flags = MAP_PRIVATE | MAP_ANONYMOUS | MAP_NORESERVE;
-      if (platform_use_hugetlb) {
-         flags |= MAP_HUGETLB;
-      }
+   int prot  = PROT_READ | PROT_WRITE;
```

Comment:
Blocks of code are indented to the right now as we've lost L 66 `if()` condition. Function will return early on L73 if `bh == NULL`.